lib/temporal/client.rb

```diff
@@ -20,31 +20,18 @@ class Client
     # @param namespace [String] Namespace to use for client calls.
     # @param interceptors [Array<Temporal::Interceptor::Client>] List of interceptors for
     #   intercepting client calls. Executed in their original order.
-    # @param payload_converter [Temporal::PayloadConverter::Base] A custom payload converter for
-    #   converting Ruby values to/from protos. See {Temporal::PayloadConverter::Base} for the
-    #   interface definition.
-    # @param payload_codecs [Array<Temporal::PayloadCodec::Base>] A list of payload codecs to
-    #   transform payload protos. See {Temporal::PayloadCodec::Base} for the interface definition.
-    # @param failure_converter [Temporal::FailureConverter::Base] A custom failure converter for
-    #   converting Exceptions to/from protos. See {Temporal::FailureConverter::Base} for the
-    #   interface definition.
+    # @param data_converter [Temporal::DataConverter] Data converter to use for all data conversions
+    #   to/from payloads.
     #
     # @see https://docs.temporal.io/concepts/what-is-a-data-converter for more information on
     #   payload converters and codecs.
     def initialize(
       connection,
       namespace,
       interceptors: [],
-      payload_converter: Temporal::PayloadConverter::DEFAULT,
-      payload_codecs: [],
-      failure_converter: Temporal::FailureConverter::DEFAULT
+      data_converter: Temporal::DataConverter.new
```

Comment:
```suggestion
      data_converter: Temporal::DataConverter::DEFAULT
```

Or something similar. Seems to make sense for consistency to have a default data converter and just reuse that like other converters (even though obviously there aren't hardly any perf gains).